scripts/test-tls12-point-format-extension.py

```diff
@@ -0,0 +1,340 @@
+    node = node.add_child(ClientKeyExchangeGenerator())
+    node = node.add_child(ChangeCipherSpecGenerator())
+    node = node.add_child(FinishedGenerator())
+    node = node.add_child(ExpectChangeCipherSpec())
+    node = node.add_child(ExpectFinished())
+    node = node.add_child(ApplicationDataGenerator(
+        bytearray(b"GET / HTTP/1.0\n\n")))
+    node = node.add_child(ExpectApplicationData())
+    node = node.add_child(AlertGenerator(AlertLevel.warning,
+                                         AlertDescription.close_notify))
+    node = node.add_child(ExpectAlert())
+    node.next_sibling = ExpectClose()
+    node.add_child(Close())
+    conversations["no_format_ext"] = conversation
+
+    # A handshake with all format extensions should pass.
+    conversation = Connect(host, port)
+    node = conversation
+    node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext_all))
+    node = node.add_child(ExpectServerHello())
+    node = node.add_child(ExpectCertificate())
+    node = node.add_child(ExpectServerKeyExchange())
+    node = node.add_child(ExpectServerHelloDone())
+    node = node.add_child(ClientKeyExchangeGenerator())
+    node = node.add_child(ChangeCipherSpecGenerator())
+    node = node.add_child(FinishedGenerator())
+    node = node.add_child(ExpectChangeCipherSpec())
+    node = node.add_child(ExpectFinished())
+    node = node.add_child(ApplicationDataGenerator(
+        bytearray(b"GET / HTTP/1.0\n\n")))
+    node = node.add_child(ExpectApplicationData())
+    node = node.add_child(AlertGenerator(AlertLevel.warning,
+                                         AlertDescription.close_notify))
+    node = node.add_child(ExpectAlert())
+    node.next_sibling = ExpectClose()
+    node.add_child(Close())
+    conversations["all_formats"] = conversation
+
+    # A handshake with only ANSI extensions should fail.
+    #
+    # See this comment in RFC 8422:
+    # > If the client sends the extension and the extension does not contain
+    # > the uncompressed point format, and the client has used the Supported "
+    # > Groups extension to indicate support for any of the curves defined in "
+    # > this specification, then the server MUST abort the handshake and "
+    # > return an illegal_parameter alert.
+    #
+    # While we _shouldn't_ see other point formats, if we do and we
+    # expect both client and server to conform to RFC 8422 (assumed
+    # by this test suite), we should be able to negotiate the
+    # uncompressed point format.
+    #
+    # However, it might also be valid to fail this request. This depends
+    # on whether the server assumes a RFC 8422 client. In particular, OpenSSL
+    # will send back a reply with all three point types.
+    conversation = Connect(host, port)
+    node = conversation
+    node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext_ansi))
+    node = node.add_child(ExpectAlert())
+    node.next_sibling = ExpectClose()
+    node.add_child(Close())
```

Comment:
This construction means: expect an Alert, or connection close; if an Alert is received, close the connection.
Shouldn't we rather wait for Alert followed by close? i.e. there should be two `ExpectClose()` nodes.